pkg/ruler/lifecycle.go

```diff
@@ -1,16 +1,28 @@
 package ruler
 
 import (
-	"context"
+	"github.com/cortexproject/cortex/pkg/ring"
 )
 
-// TransferOut is a noop for the ruler
-func (r *Ruler) TransferOut(ctx context.Context) error {
-	return nil
+func (r *Ruler) OnRingInstanceRegister(_ *ring.BasicLifecycler, ringDesc ring.Desc, instanceExists bool, instanceID string, instanceDesc ring.IngesterDesc) (ring.IngesterState, ring.Tokens) {
+	// When we initialize the store-gateway instance in the ring we want to start from
+	// a clean situation, so whatever is the state we set it JOINING, while we keep existing
+	// tokens (if any) or the ones loaded from file.
```

Comment:
The `NewTokensPersistencyDelegate()` is not used, so you should remove "or the ones loaded from file" because loading tokens from file is not supported by the ruler.

Reply:
👍 Gotcha, I just copied this block and didn't read the comments very closely. It actually still references the store-gateway